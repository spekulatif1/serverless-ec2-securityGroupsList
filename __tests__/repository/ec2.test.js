'use strict'

const {
	EC2Client,
	DescribeSecurityGroupsCommand
} = require("@aws-sdk/client-ec2");
const { mockClient } = require("aws-sdk-client-mock");
const ec2Mock = mockClient(EC2Client);

const { ec2 } = require('../../repository');
const { awsSecurityGroupsListResponse } = require('../../data/sample');

describe('ec2.repository', () => {
	afterEach(jest.clearAllMocks);

	test('ec2.repository.securityGroups.success', async () => {
		ec2Mock.on(DescribeSecurityGroupsCommand)
			.resolves(awsSecurityGroupsListResponse);

		const successResponse = await ec2.getSecurityGroups();
		expect(successResponse).toMatchSnapshot();
	});
});
