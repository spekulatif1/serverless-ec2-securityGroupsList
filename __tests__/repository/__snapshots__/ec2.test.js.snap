// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ec2.repository ec2.repository.securityGroups.success 1`] = `
Object {
  "$metadata": Object {
    "attempts": 1,
    "httpStatusCode": 200,
    "requestId": "test.request.id",
    "totalRetryDelay": 0,
  },
  "SecurityGroups": Array [
    Object {
      "Description": "TrendMicro-Assignment-Test",
      "GroupId": "sg-trendMicro",
      "GroupName": "TrendMicro-Assignment-Test",
      "IpPermissions": Array [
        Object {
          "FromPort": 80,
          "IpProtocol": "tcp",
          "IpRanges": Array [
            Object {
              "CidrIp": "0.0.0.0/0",
            },
          ],
          "Ipv6Ranges": Array [
            Object {
              "CidrIpv6": "::/0",
            },
          ],
          "PrefixListIds": Array [],
          "ToPort": 80,
          "UserIdGroupPairs": Array [],
        },
      ],
      "IpPermissionsEgress": Array [
        Object {
          "IpProtocol": "-1",
          "IpRanges": Array [
            Object {
              "CidrIp": "0.0.0.0/0",
            },
          ],
          "Ipv6Ranges": Array [],
          "PrefixListIds": Array [],
          "UserIdGroupPairs": Array [],
        },
      ],
      "OwnerId": "TrendMicro",
      "VpcId": "vpc-trendMicro",
    },
  ],
}
`;
